__tests__/app.test.js

```diff
@@ -20,12 +20,63 @@ describe("GET/api/tpics", () => {
                 });
             });
     });
-    test("404: responds with an error when the path is invalid", () => {
+    test("404: responds with 404 for non-existent routes", () => {
         return request(app)
-            .get("/api/not-a-path")
-            .expect(404)
-            .then(({ body }) => {
-                expect(body.msg).toBe("Not Found");
-            });
-    });
+          .get('/api/nonexistent')
+          .expect(404)
+          .then(({ body }) => {
+            expect(body).toHaveProperty('message', 'Endpoint not found');
+          });
+      });
 });
+describe("GET /api", () => {
+    test("200: responds with API documentation", () => {
+      return request(app)
+        .get('/api')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body).toHaveProperty('GET /api');
+          expect(body).toHaveProperty('GET /api/topics');
+          expect(body).toHaveProperty('GET /api/articles');
+
+          expect(body['GET /api']).toEqual({
+            description: "serves up a json representation of all the available endpoints of the api"
+          });
+
+          expect(body['GET /api/topics']).toEqual({
+            description: "serves an array of all topics",
+            queries: [],
+            exampleResponse: {
+              topics: [{ "slug": "football", "description": "Footie!" }]
+            }
+          });
+
+          expect(body['GET /api/articles']).toEqual({
+            description: "serves an array of all articles",
+            queries: ["author", "topic", "sort_by", "order"],
+            exampleResponse: {
+              articles: [
+                {
+                  title: "Seafood substitutions are increasing",
+                  topic: "cooking",
+                  author: "weegembump",
+                  body: "Text from the article..",
+                  created_at: "2018-05-30T15:59:13.341Z",
+                  votes: 0,
+                  comment_count: 6
+                }
+              ]
+            }
+          });
+        });
+    });
+
+    test("404: responds with 404 for non-existent routes", () => {
+      return request(app)
+        .get('/api/nonexistent')
```

Comment:
This is the same test as one you have in your first endpoint (see my comment on that pull request), it's therefore not necessary to duplicate it here again. 👍